### ISSUE TITLE
Joomla performance issue with large number of items

### ISSUE THREAD

**Issue by trananhmanh89** [joomla/joomla-cms#42313]
### Steps to reproduce the issue
1. Utilize a VPS with 4 GB of RAM and 2 CPU Cores.
2. Create a website with approximately 20,000 articles.
3. Add several Articles - Category modules to the frontend.
4. You can access my backup here https://drive.google.com/file/d/1TUBXrxGEbKNKjaakbk4YfLGgnWxN_wEq/view?usp=sharing 

### Expected result
- Article listing page speed should be fine
- Home page speed should be fine

### Actual result
- It takes around 10 seconds to load the article listing in the backend.
- Loading the website's homepage takes approximately 20 seconds.

### System information (as much as possible)
- Hawkhost vps 4G Ram, 2 CPU Cores

### Additional comments
- The performance issue is mainly due to selecting too many fields in the listing query, especially `introtext` and `fulltext`.
- After modifying the query to select only the `id` field, the speed significantly increased.
- While upgrading the server is an option, I believe Joomla can be optimized further to address this performance concern.

**Comment by richard67** [comment]
The `fulltext` column had once been added with PR #28384 .

**Comment by trananhmanh89** [comment]
> The `fulltext` column had once been added with PR #28384 .

I see, but not only the backend, the frontend is slow too.

**Comment by mavrosxristoforos** [comment]
I just unpacked your backup on my localhost. I know it's not a web server with limitations, but I just wanted to share my results:
Homepage: 225ms average loading time
Back-end articles page with 20 items: 1.10s average loading time

Is it possible that anything else on your configuration may be slowing it down?

**Comment by mavrosxristoforos** [comment]
I found your test component, so I ran it to add a few more articles. I let it add just over 50 articles. Interestingly enough, the homepage loading time went up to ~2.5s. I deleted them, but the homepage is still slow now.

**Comment by trananhmanh89** [comment]
With a sufficiently powerful PC, for example, with a Core i5 12th generation, M.2 SSD, and 16GB of RAM, the performance will be satisfactory after several refreshes (F5). This implies that you would need to invest in a VPS costing at least $40 per month to achieve relatively smooth performance with 20,000 articles. However, I don't consider this to be an ideal solution.

And this is the result on my live site, hawkhost vps 8Gb ram, 4 core (8GB LAX package).

![image](https://github.com/joomla/joomla-cms/assets/20571336/239bcee2-4907-49d2-9460-4c3ce489b29e)

**Comment by mavrosxristoforos** [comment]
What happens if you make a backup of your live site and try on localhost? Is the loading time the same?
What I'm trying to figure out here is why did it load well for me when I downloaded your backup and why doesn't it now.

In my case, the three queries that take up most of the loading time are three consecutive queries to the `#__content` with a lot of columns on `BaseDatabaseModel.php:164`. They take 900ms, 770ms and 770ms.

**Comment by trananhmanh89** [comment]
> What happens if you make a backup of your live site and try on localhost? Is the loading time the same? What I'm trying to figure out here is why did it load well for me when I downloaded your backup and why doesn't it now.
> 

This might because mysql cache your query result on ram. But sometimes it does, sometimes not. 

> In my case, the three queries that take up most of the loading time are three consecutive queries to the `#__content` with a lot of columns on `BaseDatabaseModel.php:164`. They take 900ms, 770ms and 770ms.

You can comment out these code on file `administrator\components\com_content\src\Model\ArticlesModel.php` to get better speed.
![image](https://github.com/joomla/joomla-cms/assets/20571336/1ccb6822-8aec-4cf4-9e06-ff19a7829546)

**Comment by mavrosxristoforos** [comment]
> This might because mysql cache your query result on ram. But sometimes it does, sometimes not. And at my local, the backend article list gets even more time to load, it takes 8s to load.

Yup, that was it. After restarting MySQL, I have a homepage with 225ms loading time.

I added some articles using your test component, again. Homepage > 2.5s. Restart MySQL server > 225ms.

**Comment by mavrosxristoforos** [comment]
> You can comment out these code on file `administrator\components\com_content\src\Model\ArticlesModel.php` to get better speed. 

Commenting the lines you mentioned drops the back-end loading time from 1.25s to 450ms and does not seem to affect the functionality otherwise.

The `content:` search functionality still works, and a simple search for `introtext` and `fulltext` in the back-end reveals no further uses of these columns in the list view, as far as I can tell. I would suggest you make a PR so that others can check as well.

**Comment by trananhmanh89** [comment]
@mavrosxristoforos those code was added since 4.0. I think we have to check carefully before make any PR

https://github.com/joomla/joomla-cms/commit/8d436f800c85aeaa0a3408a68d5d381583bca6c1
https://github.com/joomla/joomla-cms/commit/cf9f0fd3ba36cd2929c1e10316d3ed49c9b392ce

**Comment by mavrosxristoforos** [comment]
That is correct.
I'm thinking now that despite the fact that the articles listing back-end doesn't use these two columns directly in the list view, any extension that loads articles using the `ArticlesModel` will probably require them.
